evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -88,3 +93,57 @@ def score(self, X, y, objectives, X_train=None, y_train=None):
             y_pred_proba=None,
             objectives=objectives,
         )
+
+    def get_forecast_periods(self, X):
+        """Generates all possible forecasting periods based on last period of X.
```

Comment:
Also, a wording clarification/nitpick: how are you defining "period"? When I see period I think about a range of dates, in which case this function would return a single period. I know pandas calls them periods so that's how we're using in this case, but I'm not sure it's very clear with the wording in this function.

Reply:
Good point. My intent was to have a period = a single point of time but I guess this isn't technically true. I can change the function name to `get_forecast_period()` (singular) and say "Generates all possible forecasting time points based on last data point in X" for the docstring.

Will update with docstring example.